pipeline_dsl/concourse/task.py

```diff
@@ -95,16 +94,25 @@ def __init__(self, init_dirs, image_resource):
         self.image_resource = image_resource
 
     def package(self):
-        tar = "gtar" if platform.system() == "Darwin" else "tar"
-        files = []
-        transform = []
+        buffer = io.BytesIO()
+        tar = tarfile.open(fileobj=buffer, mode="x:bz2")
         init_dirs = sorted(list(self.init_dirs.items()), key=lambda d: len(d[1]), reverse=True)
+
+        def filter(tarinfo):
+            if not (tarinfo.isdir() or tarinfo.name.endswith(".sh") or tarinfo.name.endswith(".py")):
+                return None
+            tarinfo.uid = 0
+            tarinfo.gid = 0
+            tarinfo.uname = "root"
+            tarinfo.gname = "root"
+            tarinfo.mtime = 0
+            return tarinfo
+
         for dir_concourse, dir_local in init_dirs:
             dir_local = os.path.abspath(dir_local)
-            transform.append(f"--transform 's|{dir_local}|{dir_concourse}|g'")
-            files = files + list(glob.glob(os.path.join(dir_local, "**", "*.[ps][yh]"), recursive=True))
-        cmd = f'{tar} cj --sort=name --mtime="UTC 2019-01-01" {" ".join(transform)} --owner=root:0 --group=root:0 -b 1 -P -f - {" ".join(files)}'
```

Comment:
I thought about that, but the python documentation says:
> `TarFile.add(name, arcname=None, recursive=True, *, filter=None)`
> [...]
> Changed in version 3.7: Recursion adds entries in sorted order.
>
> Source: https://docs.python.org/3/library/tarfile.html#tarfile.TarFile.add

Currently, we only require Python >=3.6:
https://github.com/SAP/pipeline-dsl/blob/a69478d08ab15b10ed232193ca5e21fdc576a10b/setup.py#L24

But [that version is already in "security fixes only" mode,](https://www.python.org/dev/peps/pep-0494/) so we can drop it soon.

Reply:
So should we open a PR to require at least python `3.7`?

Reply:
I don't think we need to. The UX won't be as good, but it will work.
The only problem is, that it might cause unnecessary diff output when doing `fly set-pipeline` especially when calling it from different machines (`os.listdir()` will produce a rather stable order when called twice on the same machine, even if it's not guaranteed).